src/core/hle/service/mii/mii.cpp

```diff
@@ -18,35 +20,260 @@ class IDatabaseService final : public ServiceFramework<IDatabaseService> {
             {22, nullptr, "SetInterfaceVersion"},
             {23, nullptr, "Convert"},
         };
         // clang-format on
 
         RegisterHandlers(functions);
     }
+
+private:
+    template <typename OutType>
+    std::vector<u8> SerializeArray(OutType (MiiManager::*getter)(u32) const, u32 offset,
+                                   u32 requested_size, u32& read_size) {
+        read_size = std::min(requested_size, db.Size() - offset);
+
+        std::vector<u8> out(read_size * sizeof(OutType));
+
+        for (u32 i = 0; i < read_size; ++i) {
+            const auto obj = (db.*getter)(offset + i);
+            std::memcpy(out.data() + i * sizeof(OutType), &obj, sizeof(OutType));
+        }
+
+        return out;
+    }
+
+    void IsUpdated(Kernel::HLERequestContext& ctx) {
+        IPC::RequestParser rp{ctx};
+        const auto unknown{rp.PopRaw<u32>()};
+
+        LOG_WARNING(Service_Mii, "(STUBBED) called with unknown={:08X}", unknown);
+
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push(false);
+    }
+
+    void IsFullDatabase(Kernel::HLERequestContext& ctx) {
+        LOG_DEBUG(Service_Mii, "called");
+
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push(db.Full());
+    }
+
+    void GetCount(Kernel::HLERequestContext& ctx) {
+        IPC::RequestParser rp{ctx};
+        const auto unknown{rp.PopRaw<u32>()};
+
+        LOG_DEBUG(Service_Mii, "called with unknown={:08X}", unknown);
+
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u32>(db.Size());
+    }
+
+    // Gets Miis from database at offset and index in format MiiInfoElement
+    void Get(Kernel::HLERequestContext& ctx) {
+        IPC::RequestParser rp{ctx};
+        const auto size{rp.PopRaw<u32>()};
+
+        LOG_DEBUG(Service_Mii, "called with size={:08X}, offset={:08X}", size, offsets[0]);
+
+        u32 read_size{};
+        ctx.WriteBuffer(SerializeArray(&MiiManager::GetInfoElement, offsets[0], size, read_size));
+        offsets[0] += read_size;
+
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u32>(read_size);
+    }
+
+    // Gets Miis from database at offset and index in format MiiInfo
+    void Get1(Kernel::HLERequestContext& ctx) {
+        IPC::RequestParser rp{ctx};
+        const auto size{rp.PopRaw<u32>()};
+
+        LOG_DEBUG(Service_Mii, "called with size={:08X}, offset={:08X}", size, offsets[1]);
+
+        u32 read_size{};
+        ctx.WriteBuffer(SerializeArray(&MiiManager::GetInfo, offsets[1], size, read_size));
+        offsets[1] += read_size;
+
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u32>(read_size);
+    }
+
+    void BuildRandom(Kernel::HLERequestContext& ctx) {
+        IPC::RequestParser rp{ctx};
+        const auto random_params{rp.PopRaw<RandomParameters>()};
```

Comment:
Assuming the error code `0x27E` is converted to a result code like:

```cpp
constexpr ResultCode ERR_INVALID_ARGUMENT{ErrorModule::Mii, 1};
```
then the following error cases can be handled:

```suggestion
        const auto [unknown1, unknown2, unknown3] = rp.PopRaw<RandomParameters>();
        
        if (unknown1 > 2) {
            IPC::ResponseBuilder rb{ctx, 2};
            rb.Push(ERR_INVALID_ARGUMENT);
            LOG_ERROR(Service_Mii, "Invalid unknown1 value: {}", unknown1);
            return;
        }
        
        if (unknown2 > 1) {
            IPC::ResponseBuilder rb{ctx, 2};
            rb.Push(ERR_INVALID_ARGUMENT);
            LOG_ERROR(Service_Mii, "Invalid unknown2 value: {}", unknown2);
            return;
        }
        
        if (unknown3 > 2) {
            IPC::ResponseBuilder rb{ctx, 2};
            rb.Push(ERR_INVALID_ARGUMENT);
            LOG_ERROR(Service_Mii, "Invalid unknown3 value: {}", unknown3);
            return;
        }
```